service/worker/archiver/history_blob_downloader.go

```diff
@@ -62,6 +63,12 @@ type (
 	}
 )
 
+var (
+	errArchivalTokenCorrupted      = &gen.BadRequestError{Message: "Next page token is corrupted."}
+	errRequestedHistoryNotArchived = &gen.BadRequestError{Message: "Requested workflow history has not been archived yet."}
```

Comment:
I know these are two different errors and we understand how they are different but I suspect the difference is not meaningful to most people reading the errors. How about we combine these errors into one error?